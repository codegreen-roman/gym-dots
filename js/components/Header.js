import React from 'react'
import { connect } from 'react-redux'
import { shape, func, string } from 'prop-types'
import Text from './Text'
import ToggleMenu from './ToggleMenu'

const _Header = ({ exerciseSession, pingSession }) => (
    <section>
        <h4>I am a header</h4>
        <ToggleMenu />
        <p><Text text='greeting' /></p>
        <div>{exerciseSession.day}</div>
        <a href='#' onClick={pingSession}>ping</a>
    </section>
)

_Header.propTypes = {
    exerciseSession: shape({
        day: string
    }),
    pingSession: func
}

const mapStateToProps = ({ exerciseSession }) => ({ exerciseSession })

const mapActionsToProps = dispatch => ({
    pingSession() {
        dispatch({ type: 'PING' })
    }
})

export const Header = connect(mapStateToProps, mapActionsToProps)(_Header)
