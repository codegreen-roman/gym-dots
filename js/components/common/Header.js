import React from 'react'
import { connect } from 'react-redux'
import { shape, func, string } from 'prop-types'
import { Text } from '../Text'
import { ToggleMenu } from './ToggleMenu'
import { creators } from '../../state/actions'
import R from 'ramda'

const _Header = ({ exerciseSession, pingSession }) => (
    <section style={{ border: '2px solid #32CD32' }}>
        <h4>I am a header</h4>
        <ToggleMenu />
        <p><Text text='greeting' /></p>
        <div>{exerciseSession.day}</div>
        <a href='#' onClick={pingSession}>ping</a>
    </section>
)

_Header.propTypes = {
    exerciseSession: shape({
        day: string
    }),
    pingSession: func
}

const mapStateToProps = ({ exerciseSession }) => ({ exerciseSession })

const mapActionsToProps = dispatch => ({
    pingSession() {
        const app = R.compose(dispatch, creators.ping)
        app({})
    }
})

export const Header = connect(mapStateToProps, mapActionsToProps)(_Header)
