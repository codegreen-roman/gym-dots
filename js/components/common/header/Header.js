import React from 'react'
import { UserImage } from './UserImage'
import { string, object, func } from 'prop-types'
import { isEmpty } from 'ramda'


export const Header = ({ dateStr, subTitle, loginWith, auth, exerciseName, loginGuest, logout }) => {

    const rootStyle = {
        padding: '1rem',
        display: 'flex',
        justifyContent: 'space-between',
        borderBottom: '1px solid rgba(0, 0, 0, 0.15)',
        fontFamily: 'sans-serif'
    }

    const { user } = auth
    const renderTitle = () => isEmpty(exerciseName) ? subTitle : exerciseName

    const renderLoginOrUser = () => {

        const { user } = auth

        if (user) {
            return (
                <section>
                    <div className='username'>
                        <span>{user.displayName || user.uid}</span>
                    </div>
                    <button className='logout' onClick={logout}>logout</button>
                </section>
            )
        }

        return (
            <div>
                <button className='twitter' onClick={() => loginWith('twitter')}>login with twitter</button>
                <button className='facebook' onClick={() => loginWith('facebook')}>login with facebook</button>
                <button className='guest' onClick={loginGuest}>login as guest</button>
            </div>
        )
    }

    return (
        <section style={rootStyle}>

            <div className='left-side' style={{
                display: 'flex',
                flexDirection: 'column',
                justifyContent: 'space-between'
            }}>
                <a className='h-menu' style={{ fontSize: '16px' }}>&#9776;</a>
                <div className='h-date' style={{ fontSize: '12px', marginLeft: '4rem' }}>
                    {dateStr}
                </div>
                <div className='h-title' data-test='currently' style={{ fontSize: '16px', marginLeft: '3rem' }}>
                    {renderTitle()}
                </div>
            </div>

            <div className='right-side'>
                {renderLoginOrUser()}
                <UserImage image={user && user.photoURL || undefined} />
            </div>

        </section>
    )
}

Header.propTypes = {
    auth: object.isRequired,
    loginWith: func.isRequired,
    loginGuest: func.isRequired,
    logout: func.isRequired,
    dateStr: string.isRequired,
    subTitle: string.isRequired,
    exerciseName: string.isRequired
}

Header.defaultProps = {
    exerciseName: ''
}
