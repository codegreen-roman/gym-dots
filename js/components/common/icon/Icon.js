import React from 'react'
import { string } from 'prop-types'
import { svgStyle, pathStyle } from './Icon.glamor'

const icons = {
    menu:'M14 17h-10v2h10v-2zm6-8h-16v2h16v-2zm-16 6h16v-2h-16v2zm0-10v2h16v-2h-16z',
    facebook:'M6.491 19.991v-9.117H9.44l.442-3.557h-3.39V5.049c0-1.033.272-1.73 1.7-1.73H10V.14C9.686.097 8.607 0 7.358 0 4.74 0 2.957 1.66 2.957 4.695v2.622H0v3.557h2.957V20H6.49v-.009z',
    twitter:'M20 1.911a8.23 8.23 0 0 1-2.356.643A4.1 4.1 0 0 0 19.448.298a8.3 8.3 0 0 1-2.607.99A4.102 4.102 0 0 0 13.846 0C11.58 0 9.743 1.827 9.743 4.079c0 .319.036.63.106.93A11.67 11.67 0 0 1 1.393.745a4.072 4.072 0 0 0 1.27 5.446 4.12 4.12 0 0 1-1.858-.513v.051c0 1.976 1.415 3.625 3.29 4a4.178 4.178 0 0 1-1.08.143 3.96 3.96 0 0 1-.773-.076 4.102 4.102 0 0 0 3.832 2.834 8.261 8.261 0 0 1-5.095 1.743c-.332 0-.658-.02-.979-.056a11.645 11.645 0 0 0 6.289 1.836c7.547 0 11.673-6.216 11.673-11.607l-.014-.528A8.168 8.168 0 0 0 20 1.91',
    guest: 'M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z',
    dumbbells:'M33.48 6.796c-.353-.444-.85-.698-1.364-.698h-.773v-.653c0-1.173-.973-2.128-2.169-2.128h-1.018V2.112a2.15 2.15 0 0 0-.633-1.519c-.8-.79-2.183-.79-2.983 0a2.151 2.151 0 0 0-.634 1.52v5.654H10.094V2.113A2.15 2.15 0 0 0 9.46.594c-.8-.791-2.183-.791-2.983 0a2.149 2.149 0 0 0-.633 1.519v1.205H4.7a2.06 2.06 0 0 0-1.43.571c-.39.376-.615.9-.615 1.437v.772H1.89c-.516 0-1.016.255-1.37.7A2.516 2.516 0 0 0 0 8.362v3.257c0 .585.185 1.14.522 1.563.354.446.853.701 1.37.701h.764v.772c0 .537.224 1.06.616 1.437.385.368.892.571 1.43.571h1.142v1.205c0 .568.23 1.121.633 1.519.4.395.93.613 1.492.613a2.11 2.11 0 0 0 1.491-.613 2.15 2.15 0 0 0 .634-1.519v-5.653h13.812v5.653c0 .568.231 1.121.633 1.519.4.395.93.613 1.492.613a2.11 2.11 0 0 0 1.492-.613 2.15 2.15 0 0 0 .633-1.519v-1.205h1.018c1.196 0 2.17-.954 2.17-2.127v-.653h.772c.513 0 1.01-.255 1.364-.698a2.5 2.5 0 0 0 .52-1.557V8.353c0-.583-.185-1.136-.52-1.557zM1.89 12.77c-.202 0-.399-.107-.553-.3a1.384 1.384 0 0 1-.276-.852V8.362c0-.321.1-.632.276-.852.154-.193.35-.3.553-.3h.765v5.56H1.89zm2.81 2.78c-.267 0-.527-.102-.71-.278a.854.854 0 0 1-.272-.618V5.326c0-.23.096-.45.27-.617.186-.178.445-.28.712-.28h1.143v11.122H4.7zm4.33 2.317a.993.993 0 0 1-.3.71c-.398.394-1.126.394-1.524 0a.992.992 0 0 1-.3-.71V2.113c0-.267.106-.52.3-.71a1.091 1.091 0 0 1 1.524 0c.193.19.3.443.3.71v15.756zm1.063-6.765V8.878h13.812v2.225H10.094zm17 6.765a.993.993 0 0 1-.3.71c-.4.394-1.127.394-1.525 0a.992.992 0 0 1-.3-.71V2.112a.99.99 0 0 1 .3-.709 1.091 1.091 0 0 1 1.524 0c.194.19.3.443.3.71v15.755zm3.187-3.332c0 .56-.496 1.015-1.107 1.015h-1.018V4.43h1.018c.61 0 1.107.456 1.107 1.016v9.091zm2.656-2.909c0 .32-.1.627-.273.846-.153.192-.348.298-.548.298h-.773V7.21h.773c.2 0 .395.106.548.298.174.218.273.526.273.845v3.274z',
    check:'M13.5 0c1.863 0 3.617.352 5.26 1.055a13.547 13.547 0 0 1 4.298 2.887 13.547 13.547 0 0 1 2.887 4.298C26.648 9.883 27 11.637 27 13.5s-.352 3.617-1.055 5.26a13.547 13.547 0 0 1-2.887 4.298 13.547 13.547 0 0 1-4.298 2.887C17.117 26.648 15.363 27 13.5 27s-3.617-.352-5.26-1.055a13.547 13.547 0 0 1-4.298-2.887 13.547 13.547 0 0 1-2.887-4.298C.352 17.117 0 15.363 0 13.5s.352-3.617 1.055-5.26a13.547 13.547 0 0 1 2.887-4.298A13.547 13.547 0 0 1 8.24 1.055C9.883.352 11.637 0 13.5 0zm0 25.339c1.635 0 3.168-.312 4.601-.936a11.921 11.921 0 0 0 3.744-2.545 12.197 12.197 0 0 0 2.531-3.757c.624-1.433.936-2.966.936-4.601 0-1.635-.312-3.168-.936-4.601a12.034 12.034 0 0 0-2.53-3.744A12.034 12.034 0 0 0 18.1 2.624C16.668 2 15.135 1.687 13.5 1.687c-1.635 0-3.168.313-4.601.937a12.034 12.034 0 0 0-3.744 2.53A12.034 12.034 0 0 0 2.624 8.9C2 10.332 1.687 11.865 1.687 13.5c0 1.635.313 3.168.937 4.601a12.197 12.197 0 0 0 2.53 3.757A11.921 11.921 0 0 0 8.9 24.403c1.433.624 2.966.936 4.601.936zm5.379-16.77a.834.834 0 0 1 .606-.263c.229 0 .431.088.607.263a.805.805 0 0 1 .237.594c0 .237-.079.435-.237.593L11.68 18.22l-.106.105a.765.765 0 0 1-.593.264.765.765 0 0 1-.593-.264L6.196 14.16a.805.805 0 0 1-.237-.593c0-.237.08-.435.237-.593a.834.834 0 0 1 .607-.264c.228 0 .43.088.606.264l3.56 3.56 7.91-7.964z'
}

export const Icon = ({ iconName, viewBox, color, ...other }) => {
    return (
        <svg
            {...svgStyle}
            viewBox={viewBox}
            {...other}
        >
            <path {...pathStyle(color)} d={icons[iconName]} />
        </svg>
    )
}

Icon.propTypes = {
    iconName: string.isRequired,
    viewBox: string,
    color: string
}

Icon.defaultProps = {
    viewBox: '0 0 24 24'
}
