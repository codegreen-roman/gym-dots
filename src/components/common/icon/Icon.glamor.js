import { css } from 'glamor'

export const svgStyle = css({
    display: 'inline-block',
    verticalAlign: 'middle'
})

export const pathStyle = color =>
    css({
        fill: color
    })
