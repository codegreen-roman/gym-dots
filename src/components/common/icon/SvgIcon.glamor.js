import glamorous from 'glamorous'
import { variables } from '../../../utils/variables'

export const StyledSvg = glamorous.svg(
    {
        display: 'inline-block',
        color: 'black',
        height: '24px',
        width: '24px',
        userSelect: 'none',
        verticalAlign: 'middle'
    },
    ({ fillColor }) => (
        fillColor
            ? { fill: fillColor }
            : { fill: variables.$color2 }
    ),
    ({ big }) => (
        big
            ? { width: '36px', height: '36px' }
            : { width: '24px', height: '24px' }
    )
)
