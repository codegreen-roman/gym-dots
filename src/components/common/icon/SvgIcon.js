import React from 'react'
import { string, node, bool } from 'prop-types'
import { curry, prop } from 'ramda'
import { StyledSvg } from './SvgIcon.glamor'
import { icons24, icons36 } from '../../../utils/icons'

/* Wrapper to create different svg icon components */
const SvgWrap = curry((path, props) => (
    <SvgIcon {...props}>
        <path d={path} />
    </SvgIcon>
))

const propOfIcons24 = key => prop(key, icons24)
const propOfIcons36 = key => prop(key, icons36)

/* 24 24 viewbox icons */
const MenuIcon = SvgWrap(propOfIcons24('menu'))
const FacebookIcon = SvgWrap(propOfIcons24('facebook'))
const TwitterIcon = SvgWrap(propOfIcons24('twitter'))
const GuestIcon = SvgWrap(propOfIcons24('guest'))

/* 36 36 viewbox icons */
const DumbbellsIcon = SvgWrap(propOfIcons36('dumbbells'))
const CheckIcon = SvgWrap(propOfIcons36('check'))

/* Base svg component */
const SvgIcon = ({ big, viewBox, children, ...other }) => (
    <StyledSvg {...other} big={big} viewBox={big ? '0 0 36 36' : viewBox}>
        {children}
    </StyledSvg>
)

SvgIcon.propTypes = {
    viewBox: string,
    big: bool,
    children: node
}

SvgIcon.defaultProps = {
    viewBox: '0 0 24 24',
    big: false
}

MenuIcon.displayName = 'MenuIcon'
FacebookIcon.displayName = 'FacebookIcon'
TwitterIcon.displayName = 'TwitterIcon'
GuestIcon.displayName = 'GuestIcon'
DumbbellsIcon.displayName = 'DumbbellsIcon'
CheckIcon.displayName = 'CheckIcon'

export {
    MenuIcon,
    FacebookIcon,
    TwitterIcon,
    GuestIcon,
    DumbbellsIcon,
    CheckIcon,
    SvgIcon,
    SvgWrap,
    propOfIcons24,
    propOfIcons36
}
