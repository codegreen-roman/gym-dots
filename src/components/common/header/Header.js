import React from 'react'
import { UserImage } from './UserImage'
import { string, func } from 'prop-types'
import { isEmpty } from 'ramda'
import {
    header,
    headerLeftSide,
    headerDate,
    headerText,
    headerMenu,
    userWrapper,
    userPic
} from './Header.glamor.js'
import { Icon } from '../icon/Icon'
import { branch, RenderNothing } from '@utils/helpers'

export const User = ({user, logout, photoURL}) => (
    <section {...userWrapper}>
        <span {...userPic} data-test='username'>{user}</span>
        <button className='logout' onClick={logout}>
            <UserImage image={photoURL} />
        </button>
    </section>
)

User.propTypes = {
    user: string,
    logout: func,
    photoURL: string
}


export const Header = ({ dateStr, subTitle, userDisplayName, exerciseName, logout, photoURL }) => {

    const renderTitle = () => isEmpty(exerciseName) ? subTitle : exerciseName

    const renderLogoutAndUser = () => branch(userDisplayName, <User user={userDisplayName} logout={logout} photoURL={photoURL} />, <RenderNothing />)

    return (
        <header {...header}>

            <div className='left-side' {...headerLeftSide}>
                <Icon
                    iconName='menu'
                    color='red'
                    viewBox='0 0 32 32'
                    width={16}
                    height={16}
                    {...headerMenu}
                />
                <div>
                    <div data-test='date' {...headerDate}>
                        {dateStr}
                    </div>
                    <div data-test='currently' {...headerText}>
                        {renderTitle()}
                    </div>
                </div>
            </div>

            <div>
                {renderLogoutAndUser()}
            </div>

        </header>
    )
}

Header.propTypes = {
    userDisplayName: string.isRequired,
    photoURL: string,
    logout: func.isRequired,
    dateStr: string.isRequired,
    subTitle: string.isRequired,
    exerciseName: string.isRequired
}

Header.defaultProps = {
    exerciseName: ''
}
