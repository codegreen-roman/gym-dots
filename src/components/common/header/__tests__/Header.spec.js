import React from 'react'
import { Header } from '../Header'
import { shallow } from 'enzyme'
import { merge } from 'ramda'
import { shallowToJson } from 'enzyme-to-json'
import { findByDataAttr } from '@utils/testUtils'
import { Icon } from '../../icon/Icon'

const defaultProps = {
    dateStr: 'Saturday, Jul 15th',
    subTitle: 'Start Workout',
    exerciseName: '',
    photoURL: null,
    userDisplayName: '',
    logout: jest.fn(),
}

const setup = (props = defaultProps) => {
    return shallow(<Header {...props} />)
}

describe('Header component', () => {

    test('has section type of the root component', () => {
        const wrapper = setup()
        expect(wrapper.type()).toBe('header')
    })

    test('has 2 divs as 1st children', () => {
        const wrapper = setup()
        expect(wrapper.find('header > div').length).toBe(2)
    })

    describe('logged in', () => {

        const props = {
            dateStr: 'Saturday, Jul 15th',
            subTitle: 'Start Workout',
            userDisplayName: 'Roman',
            logout: jest.fn(),
        }

        let wrapper
        let User

        beforeEach(() => {
            wrapper = setup(props)
            User = wrapper.find('User')
        })

        it('should have `User` component with prop user eq Roman', () => {
            expect(User.length).toBe(1)
            expect(User.props().user).toBe('Roman')
        })
    })

    describe('Left side of the header', () => {

        let wrapper
        let menuElement
        let dateElement
        let titleElement
        const currentDate = 'Saturday, Jul 15th'

        beforeAll(() => {
            wrapper = setup()
            menuElement = wrapper.find(Icon)
            dateElement =  findByDataAttr(wrapper, 'date')
            titleElement =  findByDataAttr(wrapper, 'currently')
        })

        test('has a menu icon wrapped in a tag', () => {
            expect(menuElement.length).toBe(1)
        })

        test('has dateElement with a date value Saturday, Jul 15th', () => {
            expect(dateElement.text()).toBe(currentDate)
        })

        test('has titleElement with a value Start Workout', () => {
            expect(titleElement.text()).toBe('Start Workout')
        })
    })

    describe('Exercise name exist', () => {

        const props = merge(defaultProps, { exerciseName: 'Pull Up' })
        let wrapper
        let titleElement


        beforeAll(() => {
            wrapper = setup(props)
            titleElement =  findByDataAttr(wrapper, 'currently')
        })

        it('should match the snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        })

        it('should should show title with the exerciseName = Pull Up', () => {
            expect(titleElement.text()).toBe('Pull Up')
        })

    })

    describe('Shallow snapshot', () => {
        test('matches the previous', () => {
            const component = setup()
            const tree = shallowToJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})
