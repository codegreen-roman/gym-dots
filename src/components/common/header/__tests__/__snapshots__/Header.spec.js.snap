// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Exercise name exist should match the snapshot 1`] = `
.css-1qqtemn,
[data-css-1qqtemn] {
  height: 107px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px;
  padding-left: 40px;
  -webkit-flex-flow: row wrap;
  -webkit-justify-content: space-between;
}

.css-1ulbvcw,
[data-css-1ulbvcw] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -webkit-justify-content: space-between;
}

.css-1bdqo0b,
[data-css-1bdqo0b] {
  color: #919AA7;
}

.css-1ixk24o,
[data-css-1ixk24o] {
  text-transform: uppercase;
  color: #919AA7;
  font-size: 14px;
  font-weight: 700;
}

.css-1qe4ilz,
[data-css-1qe4ilz] {
  color: #5E6B7D;
  font-size: 23px;
}

<header
  data-css-1qqtemn=""
>
  <div
    className="left-side"
    data-css-1ulbvcw=""
  >
    <Icon
      color="red"
      data-css-1bdqo0b=""
      height={16}
      iconName="menu"
      viewBox="0 0 32 32"
      width={16}
    />
    <div>
      <div
        data-css-1ixk24o=""
        data-test="date"
      >
        Saturday, Jul 15th
      </div>
      <div
        data-css-1qe4ilz=""
        data-test="currently"
      >
        Pull Up
      </div>
    </div>
  </div>
  <div>
    <RenderNothing />
    <UserImage />
  </div>
</header>
`;

exports[`Header component Shallow snapshot matches the previous 1`] = `
.css-1qqtemn,
[data-css-1qqtemn] {
  height: 107px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px;
  padding-left: 40px;
  -webkit-flex-flow: row wrap;
  -webkit-justify-content: space-between;
}

.css-1ulbvcw,
[data-css-1ulbvcw] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -webkit-justify-content: space-between;
}

.css-1bdqo0b,
[data-css-1bdqo0b] {
  color: #919AA7;
}

.css-1ixk24o,
[data-css-1ixk24o] {
  text-transform: uppercase;
  color: #919AA7;
  font-size: 14px;
  font-weight: 700;
}

.css-1qe4ilz,
[data-css-1qe4ilz] {
  color: #5E6B7D;
  font-size: 23px;
}

<header
  data-css-1qqtemn=""
>
  <div
    className="left-side"
    data-css-1ulbvcw=""
  >
    <Icon
      color="red"
      data-css-1bdqo0b=""
      height={16}
      iconName="menu"
      viewBox="0 0 32 32"
      width={16}
    />
    <div>
      <div
        data-css-1ixk24o=""
        data-test="date"
      >
        Saturday, Jul 15th
      </div>
      <div
        data-css-1qe4ilz=""
        data-test="currently"
      >
        Start Workout
      </div>
    </div>
  </div>
  <div>
    <RenderNothing />
    <UserImage />
  </div>
</header>
`;
